### PR TITLE
ENH: add overflow handling for scalar `negative` operation

### PR DESCRIPTION
**Pull request by Micky774**
Related Issues/PRs
Follow-up to #21648

Changes
Adds check to scalar `negative` for determining whether an overflow warning should be raised for signed integers. Improves testing for `negative` operator on unsigned integers.

Comments
N/A

### CODE REVIEW

File: numpy/core/src/umath/scalarmath.c.src [hunk 1]
```diff
@@ -499,17 +499,26 @@ half_ctype_power(npy_half a, npy_half b, npy_half *out)
...
 static NPY_INLINE int
 @name@_ctype_negative(@type@ a, @type@ *out)
 {
-    *out = -a;
 #if @uns@
+    *out = -a;
     return NPY_FPE_OVERFLOW;
-#else
-    return 0;
+#elif ~@uns@ && @int@
```
Comment (seberg) [inline numpy/core/src/umath/scalarmath.c.src:514]:
```suggestion
#elif @int@
```
Nit: Or just use the logical `!` rather `~`?

**Review by seberg — APPROVED** [review]
LGTM, thanks.  Comment is just a thought, so may just merge later today in either case.

**Comment by ganesh-k13** [comment]
We are making various changes which I'm guessing will be flagged by some downstream package anytime soon. Should we release note all this somewhere? I can take it up as a separate PR to capture all this if needed.

**Comment by seberg** [comment]
Thanks @Micky774, lets put this in.  Yeah, good point @ganesh-k13.  I think downstream already noticed, although it gets drowned a bit by the cast changes.
A PR that summarizes all of these scalar math (or maybe scalar math + division) changes would be great!

**Comment by ganesh-k13** [comment]
Cool! I'll start work on it. For future reference, please tag all similar changes related to overflow to https://github.com/numpy/numpy/issues/21506

**Pull request merged**